Remove html encoded spaces from service snippets (#76)

alishaevn opened the pull request:
# Story
`&nbsp;` translates to an empty space in html (and markup apparently). it is not working the same when passed to the TextBox component though. the text itself shows. this commit will filter out that text, while leaving the remaining words, if any.

- scientist-softserv/phenovista-digital-storefront#76

# Expected Behavior Before Changes
![image](https://github.com/scientist-softserv/phenovista-digital-storefront/assets/29032869/7424b673-bf92-4e0f-bcca-2133765495d4)

# Expected Behavior After Changes
![image](https://github.com/scientist-softserv/phenovista-digital-storefront/assets/29032869/140bf081-c84f-4034-b765-26e9b18b57b7)

# Screenshots
![image](https://github.com/scientist-softserv/phenovista-digital-storefront/assets/29032869/034650f9-bda6-4675-a3f9-a3dec7a136ba)

alishaevn committed:
remove html encoded spaces from service snippets

"&nbsp;" translates to an empty space in html. it is not working the same
when passed to the TextBox component though. the text itself shows. this
commit will filter out that text, while leaving the remaining words, if any.

- scientist-softserv/phenovista-digital-storefront#76

alishaevn committed:
Merge branch 'main' into 76-remove-special-characters

labradford commented on utils/helpers.js:
```diff
@@ -89,3 +89,8 @@ export const truncateDescription = (desc = '', maxLength, isOpen) => {
 }
+
+export const removeHtmlSpaces = (text) => {
+  const regex = /&[a-z]+;/g
```
I suspect this regex is going to cause unexpected bugs because the & may be in a ware description. If you are only stripping `&nbsp` why don't you just match on that?

labradford replied:
another solution perhaps is that you can find a way to read html entities?

alishaevn replied:
this will only apply to the snippet, not the description. this particular regex would catch other html entities that we may not want, but not remove all instances of an ampersand. only cases where an ampersand is followed by letters and ends with a semicolon.
![image](https://github.com/scientist-softserv/phenovista-digital-storefront/assets/29032869/2c30bdde-1379-41e1-99c1-125ea4f947cf)

alishaevn replied:
true, reading the entities is also an option. I felt this one was quicker. but can visit the other idea if time permits.

labradford replied:
maybe https://github.com/mathiasbynens/he would work?

Regex will come back to haunt you later (says someone who may have been haunted by regex)

alishaevn committed:
decode html entities

instead of using regex, use the `he` package to decode html entities from
the ware.snippet. in the event it's an empty string, we're also trimming
that value so we don't show the textbox at all if there isn't any text.

- scientist-softserv/phenovista-digital-storefront#76
